src/main/scala/org/bitcoins/core/gen/ScriptGenerators.scala

```diff
@@ -114,19 +149,28 @@ trait ScriptGenerators extends BitcoinSLogger {
     else p2shScriptSignature
   }
 
-
+  /**
+    * Generates a [[ScriptSignature]] corresponding to the type of [[ScriptPubKey]] given.
+    * Note: Does NOT generate a correct/valid signature
+    *
+    * @param scriptPubKey
+    * @return
+    */
   private def pickCorrespondingScriptSignature(scriptPubKey : ScriptPubKey): Gen[ScriptSignature] = scriptPubKey match {
     case p2pk : P2PKScriptPubKey => p2pkScriptSignature
     case p2pkh : P2PKHScriptPubKey => p2pkhScriptSignature
     case multisig : MultiSignatureScriptPubKey => multiSignatureScriptSignature
     case EmptyScriptPubKey => emptyScriptSignature
+    case cltv : CLTVScriptPubKey => pickCorrespondingScriptSignature(cltv.scriptPubKeyAfterCLTV)
```

Comment:
We need to wrap what is returned from `pickCorrespondingScriptSIgnature` into a `CLTVScriptSignature`, same goes for `CSV`